### ISSUE TITLE
Constantly Crashes - Doesn't stay open for more than half a second

### ISSUE THREAD

**Issue by GoogleCodeExporter** [max99x/pagemon-chrome-ext#76]
```
Up to today I have had no issues with this extension, but now when I open 
chrome the page monitor instantly updates me on changes made and within half a 
second crashes. When I attempt to reload the extension, it continues to crash.


I am using Chrome version 7.0.536.2 dev and on windows 7 operating system.


```

Original issue reported on code.google.com by `grundig1...@googlemail.com` on 1 Oct 2010 at 11:57

**Comment by GoogleCodeExporter** [comment]
```
Same for me, 7.0.536.2 dev - as well.
```

Original comment by `sorin.sb...@gmail.com` on 1 Oct 2010 at 12:37

**Comment by GoogleCodeExporter** [comment]
```
By "crashes", do you mean a full crash that displays the "An addon has crashed" 
infobar at the top? If so, that's a Chrome bug, and it's unlikely I can do much 
to avoid it, especially without any extra info (does it also happen on a clean 
profile?).
```

Original comment by `max99x` on 1 Oct 2010 at 2:17

**Comment by GoogleCodeExporter** [comment]
```
If by a clean profile you mean all history including cookies deleted, yes this 
does still happen. I have installed earlier versions of page monitor i.e. just 
the one with regex and not selector advanced options, and this works fine, but 
it is the selector I use more often so this isn't really an option for me. 
```

Original comment by `grundig1...@googlemail.com` on 1 Oct 2010 at 3:08

**Comment by GoogleCodeExporter** [comment]
```
By clean profile I mean running Chrome with a --user-data-dir flag pointing to 
a temporary directory. That can rule out data corruption.
```

Original comment by `max99x` on 1 Oct 2010 at 3:13

**Comment by GoogleCodeExporter** [comment]
```
At first, when opening Chrome post-upgrade, the extension noticed some changes 
on monitored pages and alerted me - then Chrome crashed after I tried to 
open/view the extension menu or open a diff page. After a few entire Chrome 
crashes, I allowed the extension to load and didn't interact with it - and then 
the extension crashed all by itself, with a little yellow bar at the top 
informing me of this.

I've managed to tame the addon a bit since - I marked all pages as seen, and it 
seems to be stable with no alerts active. I'll test to see if it crashes when 
an alert is activated.
```

Original comment by `maxw...@gmail.com` on 1 Oct 2010 at 4:55

**Comment by GoogleCodeExporter** [comment]
```
Chrome crashed entirely when I click on "View changes" and the diff started to 
load.
```

Original comment by `maxw...@gmail.com` on 1 Oct 2010 at 5:00

**Comment by GoogleCodeExporter** [comment]
```
Cannot reproduce on Chrome Stable 6.0.472.63 or Canary 7.0.540.0.
```

Original comment by `max99x` on 1 Oct 2010 at 5:20

**Comment by GoogleCodeExporter** [comment]
```
Same problem here. Seems related to the activation of accelerated compositing 
and webgl on last dev version 7.0.536.2 on Windows 7.
I disable both functions using "--disable-accelerated-compositing 
--disable-webgl" and the extension seems to work again. I have the same problem 
on canary 7.0.540, but not on stable or beta.
```

Original comment by `gandalf....@gmail.com` on 3 Oct 2010 at 8:52

**Comment by GoogleCodeExporter** [comment]
```
Very peculiar. I'm not using WebGL or doing anything graphical at all, and 
can't reproduce it on Canary/Win7. All PM tests pass on Canary (except for a 
slight selector regression, which has been reported to the Chromium devs).
```

Original comment by `max99x` on 3 Oct 2010 at 9:26

**Comment by GoogleCodeExporter** [comment]
```
I also am seeing this problem on 7.0.536.2 on Win7.
```

Original comment by `stephen....@gmail.com` on 4 Oct 2010 at 3:39

**Comment by GoogleCodeExporter** [comment]
```
Same for Chrome 7.0.536.2 on Win XP
```

Original comment by `vhm...@gmail.com` on 6 Oct 2010 at 4:08

**Comment by GoogleCodeExporter** [comment]
```
Cannot reproduce on 7.0.536.2/Win7 either. Please confirm that this also 
happens when running with a completely clean profile (with the --user-data-dir 
Chrome flag pointing to an empty/nonexistent directory).
```

Original comment by `max99x` on 6 Oct 2010 at 6:26

**Comment by GoogleCodeExporter** [comment]
```
Is there a way to extract the PageMonitor checked URLs and advanced regex 
configurations from the file system?  I'll test clean profile option but it's 
pointless without being able to reconfigure the checks.  It seems to crash 
after displaying that some pages had changed, so the problem might be with one 
of the later pages I had configured.
```

Original comment by `stephen....@gmail.com` on 6 Oct 2010 at 4:01

**Comment by GoogleCodeExporter** [comment]
```
If you can't do a normal export, you can copy the database from (on Vista/7):
C:\Users\${username}\AppData\Local\Google\Chrome\User 
Data\Default\databases\chrome-extension_pemhgklkefakciniebenbfclihhmmfcd_0
What I would like to see is whether running a clean profile without any pages 
then adding pages still causes the crash.
```

Original comment by `max99x` on 6 Oct 2010 at 4:35

**Comment by GoogleCodeExporter** [comment]
```
That was painfully tedious.  I set up a new instance with the user data profile 
location option.  I installed PageMonitor as the only extension.  Using SQLite 
Manager I opened the default instance's PageMonitor database to get access to 
the 41 pages I had configured.  I set the new instance's PageMonitor to recheck 
every 30 seconds.  Then, one by one, I visited each of the 41 pages, enabled 
monitoring, went into the advanced options, and set the regex expression (which 
about 90% use).

Everything went fine for a lone time presumably because PageMonitor seems to 
crash on notifications and for the most part nothing changes.  But, when I 
monitored page #34 which is a non-regex basic text monitor, I got a change 
notification about 15 seconds later.  The notification popped up and the 
PageMonitor extension silently crashed.

Usually, in my default instance, I get the "extension has crashed - reload?" 
bar at the top.  In this new instance, the PageMonitor button simply 
disappeared.  Even stranger, when I went into Tools -> Extensions, the 
extension shows as not even being installed.  When I restart this new instance, 
the extension is back.

This test was consistent with the fact that PageMonitor crashes during 
notification.  In my original configuration I was not using desktop 
notification.  In my new user data profile test I was using desktop 
notifications.  So, I think the problem is not in that specific part of the 
code.

Hope that helps!
```

Original comment by `stephen....@gmail.com` on 6 Oct 2010 at 5:54

**Comment by GoogleCodeExporter** [comment]
```
Thanks for going through the test. However, this sounds only more bizarre. Can 
you perhaps provide the URL of that page #34? I've experimented with normal 
pages, regex-matching, selector-matching, upgrading using an existing profile, 
running on a clean profile, enabling and disabling notifications and various 
other settings and no crashes whatsoever. I'll have to try it on a few more 
machines, but I'm close to running out of options.
```

Original comment by `max99x` on 6 Oct 2010 at 6:05

**Comment by GoogleCodeExporter** [comment]
```
Sure, the page that triggered a notification was this one:
http://clonezilla.org/download/sourceforge/stable/iso-zip-files.php
```

Original comment by `stephen....@gmail.com` on 6 Oct 2010 at 6:09

**Comment by GoogleCodeExporter** [comment]
```
No luck. Even when faking an update on that page no crashes in sight.
```

Original comment by `max99x` on 6 Oct 2010 at 6:13

**Comment by GoogleCodeExporter** [comment]
```
Comment 8 says problem goes away with --disable-accelerated-compositing.  I've 
seen application and even OS crashes in the past with different graphics driver 
version updates related to hardware acceleration.  Is it possible that only 
some systems (because of hardware and/or drivers) are having issues with 
hardware compositing that are triggered by the notification dialog?  And 
presumably your system wouldn't have the hardware/driver issues which is why 
you can't reproduce?
```

Original comment by `stephen....@gmail.com` on 6 Oct 2010 at 6:23

**Comment by GoogleCodeExporter** [comment]
```
Problem disappeared after upgrading to 7.0.544.0 dev.  The release blog 
mentioned "numerous stability fixes".  Maybe one of these was causing the 
PageMonitor crashes.
```

Original comment by `stephen....@gmail.com` on 6 Oct 2010 at 11:59

**Comment by GoogleCodeExporter** [comment]
```
That is good to hear. If anyone else who had this problem can confirm that it's 
fixed in 7.0.544.0, I'll close the issue.
```

Original comment by `max99x` on 7 Oct 2010 at 3:21

**Comment by GoogleCodeExporter** [comment]
```
Marking as done since this appears to have been fixed upstream. Will reopen if 
anyone still experiences the error with a current Chrome version.
```

Original comment by `max99x` on 29 Oct 2010 at 2:53
* Changed state: **Done**
